overlord/devicestate/firstboot.go

```diff
@@ -415,10 +418,30 @@ var loadDeviceSeed = func(st *state.State, sysLabel string) (deviceSeed seed.See
 		return assertstate.AddBatch(st, batch, nil)
 	}
 
-	if err := deviceSeed.LoadAssertions(assertstate.DB(st), commitTo); err != nil {
+	db := assertstate.DB(st)
+	if err := deviceSeed.LoadAssertions(db, commitTo); err != nil {
 		return nil, err
 	}
 
+	// if model is dangerous, check if there is auto-import.assert and import it
+	if deviceSeed.Model().Grade() == asserts.ModelDangerous {
+		seed20AssertionsLoader, ok := deviceSeed.(seed.AutoImportAssertionsLoaderSeed)
```

Comment:
Could the contents of this if statement be put into a function to adhere to seperation of concerns for functions? For instance something called `handleAutoImports` or something like that, so it's clear exactly what this snippet of code does, both for the caller but also when reading the code.

Reply:
I have moved this into a separate function now with common with it does